CONTRIBUTING.md

```diff
@@ -1,100 +1,90 @@
-      and place a hyperlink under the number.
+3. **Accepted**:
+   - The RFC is formally approved and ready for implementation.
+   - Specific tasks and milestones are outlined for execution.
+   - **Entrance criteria**:
+     - Consensus the solution is complete.
+     - Complete changes, including tests, documentation, and examples.
 
-* Push your changes to an upstream branch:
+4. **Rejected**:
+   - If the RFC is deemed unsuitable, it is rejected with clear reasoning provided.
 
-    * Make sure that each commit in the pull request will represent a
-      logical change to the code, will compile, and will pass tests.
+5. **Implemented**:
+   - Once the proposal is successfully implemented and merged into the project.
 
-    * Make sure that the pull request message contains all important 
-      information from the commit messages including which issues are
-      fixed and closed. If a pull request contains one commit only, then
-      repeating the commit message is preferred, which is done automatically
-      by GitHub when it creates the pull request.
+Each RFC stage builds on community engagement and ensures that proposed features or changes align with project goals.
+Maintainers will assign the right stage of Issues and PR's through GitHub labels.
 
-    * Rebase your local feature branch against main (`git fetch origin`,
-      then `git rebase origin/main`) to make sure you're
-      submitting your changes on top of the newest version of our code.
+## Contributing code
 
-    * When finalizing your PR (i.e., it has been approved for merging),
-      aim for the fewest number of commits that
-      make sense. That is, squash any "fix up" commits into the commit they
-      fix rather than keep them separate. Each commit should represent a
-      clean, logical change and include a descriptive commit message.
+### Contributor License Agreement (CLA)
 
-    * Push your commit to your upstream feature branch: `git push -u <yourfork> my-feature-branch`
+Before we accept any code contributions, pgai contributors need to sign the [Contributor License Agreement (CLA)](https://cla-assistant.io/timescale/pgai). By signing a CLA, we ensure that the community is free and confident in its ability to use your contributions.
 
-* Create and manage pull request:
+### Creating a Pull Request:
 
-    * [Create a pull request using GitHub](https://help.github.com/articles/creating-a-pull-request).
-      If you know a core developer well suited to reviewing your pull
-      request, either mention them (preferably by GitHub name) in the PR's
-      body or [assign them as a reviewer](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/).
+Once your RFC is accepted (and in the case you didn't open a PR yet), or for smaller changes that do not require the RFC process, you are ready to open a PR.
+Please estrictly follow the guidelines above:
```

Comment:
```suggestion
Please strictly follow the guidelines below:
```

Reply:
LOL! And here is the proof that there are still people who dont write docs exclusively using ChatGPT (note the "exclusively" here 😆 )

Reply:
Fixed BTW. thks!